### DISCUSSION TITLE
How to convert specific images with the original remaining?

### DISCUSSION THREAD

**Discussion by DynV** [ImageMagick/ImageMagick#6092] [Help] [answered]
### ImageMagick version

7.0.10-27

### Operating system, version and so on

Cygwin 3.4.6-1.x86_64 on Windows 10 Pro 22H2

### Description

Using imagemagick, how do I convert multiple images from a directory with many more images with the original remaining after the conversion similarly to the following?

> command -filetype jpeg -progressive on -compression 70% -extension jpg -output_directory [path of different directory] -files 3.png 9.png 15.png

I'm using Cygwin on Windows 10; it's not uncommon that Cygwin versions lag behind Linux ones, but in those cases usually not super far back.

Thank you kindly for your help

**Comment by snibgo** [comment]
You seem to have invented options that IM doesn't have, but you haven't explained what you want your invented options to do. If you have a file called `3.png` and want to create a file called `3.jpg` in directory `f:\my\directory` then:
```
magick 3.png f:\my\directory\3.jpg
```
Does that help?

**Reply by DynV** [reply]
> If you have a file called `3.png` and want to create a file called `3.jpg` in directory `f:\my\directory` then:
> 
> ```
> magick 3.png f:\my\directory\3.jpg
> ```

The very beginning of my description mention multiple

> ### Description
> 
> Using imagemagick, how do I convert **multiple images** [...]

then the example has **multiple files**

> > command [...] -files 3.png 9.png 15.png

**Comment by snibgo** [comment, marked as answer]
For multiple inputs:
```
magick mogrify -format jpg 3.png 9.png 15.png
```
If the input PNG files can be read, this will create 3.jpg, 9.jpg and 15.jpg. Or:
```
magick mogrify -format jpg *.png
```

**Reply by DynV** [reply]
I did `mogrify -path ../../../Pictures -format jpg screenshot-20230219-162429.png screenshot-20230219-163140.png` and it works although I'm wondering what compression was used, as in perhaps I should set it higher so the files are smaller or lower so they look better. You don't know how to set such option?

Also if you know, is there a way to prepend something to the filenames during that operation?

**Reply by fmw42** [reply]
Mogrify does not use two images (input and output) in its command line. It just has one, which is the input name and is used for the output name. Mogrify is mostly used to process multiple input images one at a time writing to the same name as the input.  The syntax is `mogrify -path ... -format XX *.suffix`. If you want to process one image, then `mogrify -path ... -format XX image.suffix`.  The *.suffix are the input images in the current directory and the outputs will use the same names and be put in the path location.

If you want to process one image, just use `convert input .... output`.

You can set the compression to whatever you want using -quality XX in your command line.

You can read the compression of your output image in IM 7 using

`magick image.jpg -format "%[quality]\n" info: `

If in IM 6, you would have to get it from looking at 

`identify -verbose image.jpg`

**Comment by snibgo** [comment]
> Mogrify does not use two images (input and output) in its command line. It just has one, which is the input name and is used for the output name.

As I showed above, `magick mogrify` can use two or more input filenames.

> ... although I'm wondering what compression was used ...

You can use "`-quality N`" where N is a number from 1 to 100. See https://imagemagick.org/script/command-line-options.php#quality

**Reply by fmw42** [reply]
I stand corrected on multiply images.
